lib/functions/general/apt-utils.sh

```diff
@@ -0,0 +1,84 @@
+#
+# SPDX-License-Identifier: GPL-2.0
+# Copyright (c) 2023 Ricardo Pardini <ricardo@pardini.net>
+# This file is a part of the Armbian Build Framework https://github.com/armbian/build/
+#
+
+# tl;dr: this function will return the version and download URL of a package.
+# it is very naive, and will only look into a few repos across Debian/Ubuntu.
+function apt_find_upstream_package_version_and_download_url() {
+	declare sought_package_name="${1}"
+
+	declare -a package_info_download_urls=()
+	declare first_letter_of_sought_package_name="${sought_package_name:0:1}"
+	declare mirror_with_slash="undetermined/"
+
+	case "${DISTRIBUTION}" in
+		Ubuntu) # try both the jammy-updates and jammy repos, use whatever returns first
+			package_info_download_urls+=("https://packages.ubuntu.com/${RELEASE}-updates/${ARCH}/${sought_package_name}/download")
```

Comment:
Could be good to know how this behaves for some of the folks in China, or the the ones that depend on a proxy.    

Only mentioning as thing to keep an eye open  for as sometimes the apt things have gotten mess for those folks in the past.

Reply:
Yes, this is far from ideal.

`packages.ubuntu.com` or `packages.debian.org` are also sometimes out-of-date, and quite often offline/slow.

the possibility here is "finding out" what is the latest version of base-files somehow else. any ideas? they vary per RELEASE and _per ARCH_.

(Yes, we could "super quick" build a minimal rootfs just to query for this info. nah).

Otherwise, having the whole `base-files` built outside of `armbian/build`, in some kind of Launchpad/OBS/mini-buildd, and just dropping the built deb into the repo, and counting on it being correctly picked up.

Reply:
Yes, this is far from ideal.

`packages.ubuntu.com` or `packages.debian.org` are also sometimes out-of-date, and quite often offline/slow.

the possibility here is "finding out" what is the latest version of base-files somehow else. any ideas? they vary per RELEASE and _per ARCH_.

(Yes, we could "super quick" build a minimal rootfs just to query for this info. nah).

Otherwise, having the whole `base-files` built outside of `armbian/build`, in some kind of Launchpad/OBS/mini-buildd, and just dropping the built deb into the repo, and counting on it being correctly picked up.

Reply:
Good point of this is that, those files are changing rarely which means they will be cached unless ARTIFACT_IGNORE_CACHE='yes' will be used.

Reply:
I have put together with adjustments of base-files into https://github.com/armbian/build/pull/5192 I am aware we need to address this, but for the purpose of generating a release, this problem should not surface. I think its better to include this then previous half solution and there is still some time to address problems if found.